main/src/libraries/wrdac/src/clients/icubClient.cpp

```diff
@@ -610,6 +621,143 @@ bool ICubClient::push(const Vector &target, const Bottle &options, std::string s
 }
 
 // KARMA
+
+// Left push
+bool ICubClient::pushKarmaLeft(const std::string &objName, const double &targetPosYLeft,

```

Comment:
Just as a general remark, you should try to avoid duplicating code. Here, the code of `pushKarmaLeft` and `pushKarmaRight` is 90% identical, except of some parameters. As WYSIWYD is in general not the best coded thing I've ever seen ..., it's fine here ;).